### PR TITLE
Update all delete mutations to return entity id

### PR DESCRIPTION
**Pull request by wongcmichelle**
[Notion ticket](https://www.notion.so/uwblueprintexecs/delete-and-edit-mutation-cleanup-98ad81733016484b928909fb184e7360)

## Changes
- Update following delete mutations to return the deleted ID
    - ambulance
    - CCP (collectionPoint)
    - event
    - hospital
    - locationPin
    - patient
    - user

## Testing
**Ambulance**
Delete Successfully
image
image
Delete Error
- Error deleting ambulance with associated patients
image
- Error deleting ambulance with invalid ID
image

**CCP**
Delete Successfully
image
image
Delete Error
- Invalid ID
image

**Event**
Delete Successfully
image
image
Delete Error
- Invalid ID
image

**Hospital**
Delete Successfully
image
image

Delete Error
- Error deleting hospital with associated patients
image
- Error deleting hospital with invalid ID
image

**Location Pin**
Delete Successfully
image
image
Delete Error
- Invalid ID
image

**Patient**
Delete Successfully
image
image
Delete Error
- Invalid ID
image

**User**
Delete Successfully
image
image
Delete Error
- Invalid ID
image

before opening PR
- [x] check notion ticket
- [x] run linter
- [x] go through file diff

filling out PR
- [x] descriptive title
- [x] update `notion ticket` link
- [x] fill out `Changes`
  - [x] \(optional) add inline comments in diff
- [x] fill out `Testing`
- [x] \(optional) add `Screenshots`
- [x] assign yourself to the PR

after opening PR
- [x] link PR to notion ticket
- [x] move ticket to `In PR`
- [x] make sure build passes
- [x] ping `@ps` in `#paramedics-dev`

### CODE REVIEW

File: resolvers/ambulance.js
Comment (wongcmichelle) [inline resolvers/ambulance.js:91]: `destroy` returns a Promise that always gets resolved with `1` on success or `0` otherwise (i.e. the promise is always “fulfilled” and won’t trigger a `catch` statement event when the deleting failed)

File: resolvers/event.js
Comment (wongcmichelle) [inline resolvers/event.js:309]: destroy returns a Promise that always gets resolved with 1 on success or 0 otherwise (i.e. the promise is always “fulfilled” and won’t trigger a catch statement event when the deleting failed).

Note this means that there's no "error handling" for `db.eventAmbulances.destroy` and `db.eventHospitals.destroy`. We can implement it but I couldn't find a clean way to do it.

File: resolvers/ambulance.js
Comment (wongcmichelle) [inline resolvers/ambulance.js:90]: We have no way of knowing why it failed at this point, so that's why the error message is pretty general.

File: resolvers/hospital.js
Comment (wongcmichelle) [inline resolvers/hospital.js:78]: destroy returns a Promise that always gets resolved with 1 on success or 0 otherwise (i.e. the promise is always “fulfilled” and won’t trigger a catch statement event when the deleting failed).

Note this means that there's no "error handling" for `db.eventHospitals.destroy` here.

File: resolvers/locationPin.js
Comment (petorsong) [inline resolvers/locationPin.js:83]: is there a way to generalize this and pull it out like we did for `utils/validators`? (i know it might not work well with the resource associations but this is a little repetitive)

File: resolvers/ambulance.js
Comment (petorsong) [inline resolvers/ambulance.js:58]: i'm not a fan of promise chaining especially this many times - i know i've leaned towards it before but i feel like there's been issues with readability and swallowing errors thrown inside promises; to be clear, i think one promise + one chain is reasonable, but beyond that it gets kinda messy, especially if there's not a whole lot happening inside them

i'm liking `async`/`await` more now, as it's more intuitive (maybe just for me) and easier to deal with errors - i've been  pushing for that with new devs (e.g. https://github.com/uwblueprint/paramedics-web/pull/66)

just wanted to thoughts from you and @aaronabraham311 on what pattern we should move forward with

**Review by petorsong — COMMENTED** [review]
pretty straightforward, just 2 comments i'd like addressed!

File: resolvers/locationPin.js
Comment (wongcmichelle) [reply resolvers/locationPin.js:83]: could take this part out `(isDeleted) => {
          if (isDeleted === 1) {
            return args.id;
          }
          throw new Error('Deletion failed for location pin ID: ' + args.id);
        }`

and define something like `successCallback = (isDeleted, entityTypeLabel) => {
          if (isDeleted === 1) {
            return args.id;
          }
          throw new Error(`Deletion failed for ${entityTypeLabel} ID: ${args.id}`);
        }`

File: resolvers/ambulance.js
Comment (wongcmichelle) [reply resolvers/ambulance.js:58]: I personally think it's cleaner to use promise chaining here if we're using the result returned by these promises. Since like I mentioned, `try` and `catch` wouldn't do anything for us because `destroy` always results in a fulfilled promise, but it's just a matter of whether its return value is `0` or `1`.

Comment (wongcmichelle) [reply resolvers/ambulance.js:58]: I don't think it's about whether you consistently use async/await or promise chaining, since they each have their uses https://medium.com/better-programming/should-i-use-promises-or-async-await-126ab5c98789

File: resolvers/locationPin.js
Comment (petorsong) [reply resolvers/locationPin.js:83]: Your call if you want to do that refactoring - our usage of `destroy` is not super consistent/clean so I'm ok if we leave it this way

File: resolvers/ambulance.js
Comment (petorsong) [reply resolvers/ambulance.js:58]: Thanks for the link, that does help with clarifying between the two options - in that case I'm fine with keeping this the way it is

**Review by aaronabraham311 — APPROVED** [review]
Awesome work, lgtm!

**Review by petorsong — APPROVED** [review]

File: resolvers/locationPin.js
Comment (wongcmichelle) [reply resolvers/locationPin.js:83]: @petorsong do you want this refactoring before i merge the PR?

Comment (petorsong) [reply resolvers/locationPin.js:83]: Nah it's fine, I'm good with merging as is

**Pull request merged**
